illumos-utils/src/zfs.rs

```diff
@@ -229,6 +303,38 @@ impl Zfs {
         Ok(filesystems)
     }
 
+    /// Get information about datasets within a list of zpools / datasets.
+    ///
+    /// This function is similar to [Zfs::list_datasets], but provides a more
+    /// substantial results about the datasets found.
+    ///
+    /// Sorts results and de-duplicates them by name.
+    pub fn get_dataset_properties(
+        datasets: &[String],
+    ) -> Result<Vec<DatasetProperties>, anyhow::Error> {
+        let mut command = std::process::Command::new(ZFS);
+        let cmd = command.args(&["list", "-d", "1", "-rHpo"]);
+
+        // Note: this is tightly coupled with the layout of DatasetProperties
+        cmd.arg("oxide:uuid,name,avail,used,quota,reservation,compression");
+        cmd.args(datasets);
+
+        let output = execute(cmd).with_context(|| {
+            format!("Failed to get dataset properties for {datasets:?}")
+        })?;
+        let stdout = String::from_utf8_lossy(&output.stdout);
```

Comment:
Think this should fail on non-UTF-8 output.

Reply:
Done